experimenter/tests/integration/nimbus/android/conftest.py

```diff
@@ -0,0 +1,154 @@
+    yield gradlewbuild_log
+
+
+@pytest.fixture
+def gradlewbuild(gradlewbuild_log):
+    yield GradlewBuild(gradlewbuild_log)
+
+
+@pytest.fixture(name="ping_server", autouse=True, scope="session")
+def fixture_ping_server():
+    path = next(iter(here.glob("**/android/ping_server")))
+    process = start_process(path, ["python", "ping_server.py"])
+    yield "http://localhost:5000"
+    process.terminate()
+
+
+@pytest.fixture(name="delete_telemetry_pings")
+def fixture_delete_telemetry_pings(ping_server):
+    def runner():
+        requests.delete(f"{ping_server}/pings")
+
+    return runner
+
+
+@pytest.fixture(name="check_ping_for_experiment")
+def fixture_check_ping_for_experiment(experiment_slug, ping_server, open_app):
+    def _check_ping_for_experiment(branch=None, experiment=experiment_slug, reason=None):
+        model = TelemetryModel(branch=branch, experiment=experiment)
+
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            data = requests.get(f"{ping_server}/pings").json()
+            events = []
+            for item in data:
+                event_items = item.get("events")
+                if event_items:
+                    for event in event_items:
+                        if (
+                            "category" in event
+                            and "nimbus_events" in event["category"]
+                            and "extra" in event
+                            and "branch" in event["extra"]
+                        ):
+                            events.append(event)
+            for event in events:
+                event_name = event.get("name")
+                if (reason == "enrollment" and event_name == "enrollment") or (
+                    reason == "unenrollment"
+                    and event_name in ["unenrollment", "disqualification"]
+                ):
+                    telemetry_model = TelemetryModel(
+                        branch=event["extra"]["branch"],
+                        experiment=event["extra"]["experiment"],
+                    )
+                    if model == telemetry_model:
+                        return True
+            time.sleep(5)
+            open_app()  # Open app in cycles to trigger telemetry
+        return False
+
+    return _check_ping_for_experiment
+
+
+@pytest.fixture(name="set_experiment_test_name", autouse=True, scope="session")
+def fixture_set_experiment_test_name():
+    os.environ["EXP_NAME"] = "fenix"
+
+
+@pytest.fixture(name="setup_experiment")
+def fixture_setup_experiment(
+    run_nimbus_cli_command,
+    delete_telemetry_pings,
+):
+    def _():
+        experiment_json = next(iter(here.glob("**/fixtures/experiment.json")))
+        delete_telemetry_pings()
+        logging.info("Testing fenix e2e experiment")
+        command = f"nimbus-cli --app fenix --channel developer enroll firefox-fenix-test-experiment --branch control --file {experiment_json} --reset-app"  #  NOQA
```

Comment:
You should prefer passing a list of strings instead of a single string so we dont have to worry about shell quoting.